src/main/java/jdwp/Translator.java

```diff
@@ -192,15 +215,54 @@ public static String normalizeFunc(String func) {
 				newParams.add(param);
 			}
 		}
-		String newParamList = "";
-		for (int i = 0; i < newParams.size(); i++) {
-			if (i == 0) {
-				newParamList += newParams.get(i);
-			} else {
-				newParamList += ", " + newParams.get(i);
-			}
+
+		StringBuilder newParamList = new StringBuilder();
+		for (String newParam : newParams) {
+			newParamList.append(newParam);
+		}
+
+		finalSignature.append(start);
+		finalSignature.append("(");
+		finalSignature.append(newParamList);
+		finalSignature.append(")");
+		return finalSignature.toString();
+	}
+
+	public static String getPrimitiveJNI(String param) {
+		String jniParam = "";
+
+		switch (param) {
+			case "":
+				break;
+			case JAVA_BOOLEAN:
+				jniParam = typeSignature.get(JAVA_BOOLEAN);
+				break;
+			case JAVA_BYTE:
+				jniParam = typeSignature.get(JAVA_BYTE);
+				break;
+			case JAVA_CHAR:
+				jniParam = typeSignature.get(JAVA_CHAR);
+				break;
+			case JAVA_SHORT:
+				jniParam = typeSignature.get(JAVA_SHORT);
+				break;
+			case JAVA_INT:
+				jniParam = typeSignature.get(JAVA_INT);
+				break;
+			case JAVA_LONG:
+				jniParam = typeSignature.get(JAVA_LONG);
+				break;
+			case JAVA_FLOAT:
+				jniParam = typeSignature.get(JAVA_FLOAT);
+				break;
+			case JAVA_DOUBLE:
+				jniParam = typeSignature.get(JAVA_DOUBLE);
+				break;
+			case JAVA_VOID:
+				jniParam = typeSignature.get(JAVA_VOID);
+				break;
 		}
-		return start + "(" + newParamList + ")";
+		return jniParam;
```

Comment:
@jchen10500 We could reduce this method to just be `return typeSignature.get(param);` without a switch/case clauses list...wdyt?

Reply:
:thumbsup: